chia/consensus/blockchain.py

```diff
@@ -1068,7 +1068,12 @@ async def lookup_block_generators(self, header_hash: bytes32, generator_refs: Se
         # need to find the fork point
         peak_block = await self.get_block_record_from_db(header_hash)
         assert peak_block is not None
-        if self.height_to_hash(peak_block.height) != header_hash:
+        height_hash: Optional[bytes32] = None
+        try:
+            height_hash = self.height_to_hash(peak_block.height)
+        except Exception:
```

Comment:
It seems risky to catch (and ignore) *all* exceptions. This would include assertion failures.
I think it would be more precise to either check if the current chain has the block height we're about to look up.